app/pages/ServiceDiscoveryForm/ServiceDiscoveryForm.tsx

```diff
@@ -4,63 +4,110 @@ import qs from 'qs';
+interface SelectedRefinements {
   [key: number]: boolean;
 }
 
 /** Wrapper component that handles state management, URL parsing, and external API requests. */
 export const ServiceDiscoveryForm = () => {
-  const match = useRouteMatch();
   interface MatchParams {
     categorySlug: string;
   }
 
+  const match = useRouteMatch();
   const { categorySlug } = match.params as MatchParams;
   const category = CATEGORIES.find(c => c.slug === categorySlug);
   if (!category) {
     // Category does not exist; user may have entered the category in the URL bar
-    // or there is an error in our code
     return <Redirect push to={{ pathname: '/' }} />;
   }
 
-  const eligibilities: CategoryRefinement[] = useEligibilitiesForCategory(category.id) || [];
-  const subcategories: CategoryRefinement[] = useSubcategoriesForCategory(category.id) || [];
   return (
     <InnerServiceDiscoveryForm
-      categorySlug={category.slug}
-      eligibilities={eligibilities}
-      subcategories={subcategories}
-      steps={category.steps}
+      category={category}
       subcategorySubheading={category.subcategorySubheading}
     />
   );
 };
 
 /** Main component that handles form data and advancing steps. */
 const InnerServiceDiscoveryForm = ({
-  steps, eligibilities, subcategories, categorySlug, subcategorySubheading,
+  category, subcategorySubheading,
 }: {
-  steps: Step[];
-  eligibilities: CategoryRefinement[];
-  subcategories: CategoryRefinement[];
-  categorySlug: string;
+  category: ServiceCategory;
   subcategorySubheading: string;
 }) => {
+  const {
+    steps,
+    id,
+    slug: categorySlug,
+  } = category;
+
+  // The activeCategoryId is updated if the user proceeds to a further step that has child
+  // subcategories to be displayed. When it is set, the target subcategory refinements are
+  // fetched and rendered in place of the previous refinements
+  const [activeCategoryId, setActiveCategoryId] = useState<string | null>(id);
   const [currentStep, setCurrentStep] = useState(0);
+  // Add the selectedRadioItem to state so that when a user clicks the next button
+  // the selectedRadioItem can be passed to the goToNextStep function
+  const [selectedRadioItem, setSelectedRadioItem] = useState<number | null>(null);
+  const history = useHistory();
+
+  const eligibilities: CategoryRefinement[] = useEligibilitiesForCategory(activeCategoryId)
+  || [];
+  const subcategories: CategoryRefinement[] = useSubcategoriesForCategory(activeCategoryId)
+  || [];
 
+  let goToNextStep;
+  const stepName = steps[currentStep];
+  const disableNextBtn = selectedRadioItem === null && ['housingStatus', 'subcategoriesRadio'].includes(stepName);
   // TODO: Should goBack go back to the previous step?
-  const history = useHistory();
-  const goBack = () => history.goBack();
-  const goToNextStep = () => setCurrentStep(currentStep + 1);
+  const goBack = () => {
+    history.push('/');
+  };
+
+  if (stepName === 'housingStatus') {
```

Comment:
I tried to figure out a more abstract solution than adding Long Term Housing specific conditions in the service discovery form, but I didn't feel like I had the time given our demo deadline.  I think that if we reorganize how we structure our category/steps data in the constants.ts file it might allow for more flexibility here, though that would of course include a larger refactor of this module.